app/Http/Controllers/AgentController.php

```diff
@@ -54,4 +55,24 @@ public static function createAgent($data)
 
         return Agent::query()->create($data);
     }
+
+    public function deleteAgent($id)
+    {
+        $agent = Agent::query()->find($id);
+        if (!$agent) {
+            return res('Authority not found', 404);
+        }
+
+        DB::beginTransaction();
+        try {
+            $user = User::find($agent->user_id);
+            $agent->delete();
+            $user->delete();
+
+            DB::commit();
+            return res('Agent deleted successfully');
+        } catch (Exception $e) {
+            return res($e->getMessage(), 400);
+        }
```

Comment:
Need to add => DB::rollback() here to revert your changes.

Reply:
👍